### PR TITLE
Remove custom code to disable collection delete (fixes #46)

### LINKED ISSUE #46: Refactor disabling of endpoint
I believe we should get rid of the code that disables the DELETE endpoint on collections via a specific setting.

IHMO, this should be : 
* out of cliquet scope (eg. do in readinglist endpoint override)
* applied to any endpoints through generic configuration (setting like ``{method}_{collection|record}_{resource}_enabled``)
* through options during registering of resource (see #12)

### PR DESCRIPTION
**Pull request by leplatrem**
@Natim r?

### CODE REVIEW

File: cliquet/resource.py [hunk 1]
```diff
@@ -230,19 +229,19 @@ def register_service(endpoint_type):
...
 
-    services = [register_service('collection'), register_service('record')]
-
     def callback(context, name, ob):
         # get the callbacks registred by the inner services
         # and call them from here when the @resource classes are being
         # scanned by venusian.
+        config = context.config.with_package(info.module)
+        services = [register_service('collection', config.registry.settings),
+                    register_service('record', config.registry.settings)]
```
Comment (leplatrem) [inline cliquet/resource.py]: Caution: the parameter is called ``config`` and you pass ``settings``

File: cliquet/tests/resource/test_viewset.py [hunk 2]
```diff
@@ -277,9 +310,15 @@ def test_record_views_are_registered_in_cornice(self, service_class):
...
-        register_resource(self.resource, viewset=self.viewset, settings={
+        venusian_callback = register_resource(
+            self.resource, viewset=self.viewset)
+
+        context = mock.MagicMock()
+        context.registry.settings = {
             'cliquet.record_fake_put_enabled': False
-        })
+        }
+        context.config.with_package.return_value = context
```
Comment (leplatrem) [inline cliquet/tests/resource/test_viewset.py]: This is weird, but I believe there is nothing better :|

Comment (Natim) [inline cliquet/tests/resource/test_viewset.py]: Are you sure that it works without mocks ? I mean that with_package actually returns you that?

File: cliquet/resource.py (hunk 1 above)
Comment (ametaireau) [inline cliquet/resource.py]: updated.

File: cliquet/tests/resource/test_viewset.py (hunk 2 above)
Comment (ametaireau) [inline cliquet/tests/resource/test_viewset.py]: http://pyramid.readthedocs.org/en/latest/api/config.html#pyramid.config.Configurator.with_package

**Comment by leplatrem** [comment]
We should remove/merge this : https://github.com/mozilla-services/cliquet/commit/82eae2a0253b94bfc663968a89ab1ee5be7ab539

**Comment by ametaireau** [comment]
Updated.

**Pull request merged**